mmv1/third_party/terraform/services/artifactregistry/data_source_artifact_registry_docker_image_test.go

```diff
@@ -0,0 +1,151 @@
+package artifactregistry_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccDataSourceArtifactRegistryDockerImage(t *testing.T) {
+	t.Parallel()
+
+	resourceName := "data.google_artifact_registry_docker_image.test"
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceArtifactRegistryDockerImageConfig,
+				Check: resource.ComposeTestCheckFunc(
+					// Data source using a tag
+					checkTaggedDataSources(resourceName+"Tag", "latest"),
+					resource.TestCheckResourceAttrSet(resourceName+"Tag", "image_size_bytes"),
+					validateTimeStamps(resourceName+"Tag"),
+
+					// Data source using a digest
+					checkDigestDataSources(
+						resourceName+"Digest",
+						"projects/cloudrun/locations/us/repositories/container/dockerImages/hello@sha256:77cb9fbc6a667b8bfdbeca4c49e7703d825746eba53b736f0318bb7712828821",
+						"us-docker.pkg.dev/cloudrun/container/hello@sha256:77cb9fbc6a667b8bfdbeca4c49e7703d825746eba53b736f0318bb7712828821",
+					),
+					resource.TestCheckResourceAttrSet(resourceName+"Digest", "image_size_bytes"),
+					validateTimeStamps(resourceName+"Digest"),
+
+					// url safe docker name using a tag
+					checkTaggedDataSources(resourceName+"UrlTag", "latest"),
+
+					// url safe docker name using a digest
+					checkDigestDataSources(
+						resourceName+"UrlDigest",
+						"projects/go-containerregistry/locations/us/repositories/gcr.io/dockerImages/krane%2Fdebug@sha256:26903bf659994649af0b8ccb2675b76318b2bc3b2c85feea9a1f9d5b98eff363",
+						"us-docker.pkg.dev/go-containerregistry/gcr.io/krane/debug@sha256:26903bf659994649af0b8ccb2675b76318b2bc3b2c85feea9a1f9d5b98eff363",
+					),
+
+					// Data source using no tag or digest
+					resource.TestCheckResourceAttrSet(resourceName+"None", "repository_id"),
+					resource.TestCheckResourceAttrSet(resourceName+"None", "image_name"),
+					resource.TestCheckResourceAttrSet(resourceName+"None", "name"),
+					resource.TestCheckResourceAttrSet(resourceName+"None", "self_link"),
+				),
+			},
+		},
+	})
+}
+
+// Test the data source against the public AR repos
+// https://console.cloud.google.com/artifacts/docker/cloudrun/us/container
+// https://console.cloud.google.com/artifacts/docker/go-containerregistry/us/gcr.io
+// Currently, gcr.io does not provide a imageSizeBytes or buildTime field in the JSON response
+const testAccDataSourceArtifactRegistryDockerImageConfig = `
+provider "google" {
```

Comment:
I think this test config might cause issues in our test environment:
1. You should remove the provider block
2. When run in our test environments, we won't have these projects already created. You'll want to use the primary test project (which means you shouldn't need to specify it), but then you'll need to seed test data there. Maybe you can create what you need with another TF resource (it looks like other tests have done so)?

Reply:
Yea, I can remove the provider block. I just had it to make sure that the data block for `testTag` and `testDigest` were using the project defined within a provider block.

If that's not necessary, I can remove the provider block and define the project fields for the `testTag` and `testDigest` data sources.

Reply:
Ah ok, I just checked and it does look like the block won't break things, but we have no other tests that define a provider-level default for project. I would stay consistent with those and specify project on each resource/data-source that needs it

Reply:
@roaks3 Alright, removed the provider block and updated the tests to include the project field.

Reply:
Sorry, it looks like my second point from above got lost in the mix. We don't have a `cloudrun` project already created in our test environment, so I'd expect the config to fail. You probably want to pass `envvar.GetTestProjectFromEnv()` into your test, which is the id of our primary test project, but then you'd need to add config to your test to set it up so these data sources can retrieve data.

Does that make sense? What you have here now looks like it is written against resources that you created ahead of time, and we just need to replicate that.